resource/graph_node.go

```diff
@@ -28,11 +28,13 @@ type GraphNode struct {
 	markedForRemoval          bool
 	unresolvedDependencies    []string
 	needsDependencyResolution bool
+	timesReconfigured         uint8
```

Comment:
I appreciate what I assume is an attempt to take up less memory, but the usage of `uint8` here may be confusing to future developers. Small integer types like this are usually only used when memory is a major concern (e.g. struct needs to be serialized and sent over a network a lot), or when using an `iota` enum declaration for some enum that will certainly not exceed 256 values.

You should probably just make this an `atomic.Uint64` and access it through `Store` and `Load` and `Add*`. You won't need the graph node's `mu` either which should assuage your concern about deadlocking.